File: eidocell-ui/src/types/classes.ts
export interface ClassSummary {
  id: string
  name: string
  color: string
  sample_count: number
}

export interface AttributeStatistics {
  name: string
  mean: number | null
  std: number | null
  min: number | null
  max: number | null
  median: number | null
}

export interface ClassStatistics {
  id: string
  name: string
  color: string
  sample_count: number
  attributes: AttributeStatistics[]
}

export interface AttributeDistribution {
  name: string
  bin_edges: number[]
  bin_counts: number[]
  min?: number | null
  max?: number | null
  mean?: number | null
  std?: number | null
}

export interface SessionDistributions {
  sample_count: number
  attributes: AttributeDistribution[]
}

File: eidocell-backend/schemas/workspace/classes.py
from pydantic import BaseModel


class ClassSummary(BaseModel):
    id: str
    name: str
    color: str
    sample_count: int


class ClassSamplesPage(BaseModel):
    items: list[dict]  # same shape as SampleOut
    total: int
    offset: int
    limit: int


class AttributeStatistics(BaseModel):
    name: str
    mean: float | None = None
    std: float | None = None
    min: float | None = None
    max: float | None = None
    median: float | None = None


class ClassStatistics(BaseModel):
    id: str
    name: str
    color: str
    sample_count: int
    attributes: list[AttributeStatistics]


class AttributeDistribution(BaseModel):
    name: str
    bin_edges: list[float]
    bin_counts: list[int]
    min: float | None = None
    max: float | None = None
    mean: float | None = None
    std: float | None = None


class SessionDistributions(BaseModel):
    sample_count: int
    attributes: list[AttributeDistribution]
